src/Umbraco.Web.BackOffice/DependencyInjection/UmbracoBuilder.LocalizedText.cs

```diff
@@ -42,20 +44,24 @@ private static IUmbracoBuilder AddSupplemenataryLocalizedTextFileSources(this IU
             // gets all langs files in /app_plugins real or virtual locations
             IEnumerable<LocalizedTextServiceSupplementaryFileSource> pluginLangFileSources = GetPluginLanguageFileSources(webFileProvider, Cms.Core.Constants.SystemDirectories.AppPlugins, false);
 
-            // user defined langs that overwrite the default, these should not be used by plugin creators
+            // user defined language files that overwrite the default, these should not be used by plugin creators
             var userConfigLangFolder = Cms.Core.Constants.SystemDirectories.Config
                                             .TrimStart(Cms.Core.Constants.CharArrays.Tilde);
 
-            IEnumerable<LocalizedTextServiceSupplementaryFileSource> userLangFileSources = contentFileProvider.GetDirectoryContents(userConfigLangFolder)
-                    .Where(x => x.IsDirectory && x.Name.InvariantEquals("lang"))
-                    .Select(x => new DirectoryInfo(x.PhysicalPath))
-                    .SelectMany(x => x.GetFiles("*.user.xml", SearchOption.TopDirectoryOnly))
-                    .Select(x => new LocalizedTextServiceSupplementaryFileSource(x, true));
+            var configLangFileSources = new List<LocalizedTextServiceSupplementaryFileSource>();
+
+            foreach (IFileInfo langFileSource in contentFileProvider.GetDirectoryContents(userConfigLangFolder).Where(x => x.IsDirectory && x.Name.InvariantEquals("lang")))
+            {
+                foreach (IFileInfo langFile in contentFileProvider.GetDirectoryContents($"{userConfigLangFolder}/{langFileSource.Name}").Where(x => x.Name.InvariantEndsWith(".xml") && x.PhysicalPath != null))
+                {
+                    configLangFileSources.Add(new LocalizedTextServiceSupplementaryFileSource(langFile, true));
+                }
+            }
```

Comment:
```suggestion
            foreach (IFileInfo langFileSource in contentFileProvider.GetDirectoryContents(userConfigLangFolder))
            {
                if (langFileSource.IsDirectory && langFileSource.Name.InvariantEquals("lang"))
                {
                    foreach (IFileInfo langFile in contentFileProvider.GetDirectoryContents($"{userConfigLangFolder}/{langFileSource.Name}"))
                    {
                        if(langFile.Name.InvariantEndsWith(".xml") && langFile.PhysicalPath is not null)
                        {
                            configLangFileSources.Add(new LocalizedTextServiceSupplementaryFileSource(langFile, true));
                        }
                    }
                }
            }
```

Not that I think that this has a major impact, but if we're gonna use foreach loops instead of linq (which is completely valid IMO), I think we should use it all the way, this way see save have to do 2 additional enumerations (one for each Where clause).

Reply:
Very good point